game/lib/game/assets/tileset.dart

```diff
@@ -68,6 +68,9 @@ class Tileset {
     blocks = [1, 2].map((i) => BlockSet(_sheet, i)).toList();
     planets = _sheet.generate('back-props', (key) => _sheet.sprite(key));
     stars = _sheet.generate('stars-pattern', (key) => _sheet.sprite(key));
+
+    // TODO not a good solution, but can't think of anything
+    await Future.delayed(Duration(seconds: 1));
```

Comment:
@luanpotter without this line, the code breaks inside Flame, because the sprites extracted from the SpriteSheets are not loaded yet, this happens because the constructor of Sprite are async and when the sprites are accessed, a NPE happens. I am pretty aware that this solution is ridiculous, I just wanted a quick way to get the game working on this new structure, so I could start testing.

Please, help me finding a better solution for this, reach me if you need further context about this problem.

Reply:
To be honest, probably just by adding the splash screen should solve this.

Reply:
the solution is that you need to check the isLoaded method inside your sprites